### ISSUE TITLE
Generalization issue of equalized odds (exponentiatedgradient and thresholdoptimizer)

### ISSUE THREAD

**Issue by ziqi-ma** [fairlearn/fairlearn#890]
Hello,

I ran into the problem of generalization in enforcing equalized odds - if I equalize on training set, on test set I get quite unfair (significant after bootstrapping) performance. The test data is not out of distribution. I tried both thresholdOptimizer and exponentiatedGradient, and ran into generalization issues in both. I am wondering if this is a common issue.

**Comment by romanlutz** [comment]
@ziqi-ma you may want to take a look at #460 which goes in depth about this topic. If there are additional questions feel free to reopen. Thanks!
